caddyhttp/fastcgi/fastcgi.go

```diff
@@ -157,17 +159,20 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 				return http.StatusBadGateway, err
 			}
 
+			respBody, err := ioutil.ReadAll(resp.Body)
```

Comment:
This is a little unsafe, what if the response body is a large file download (GBs in size)?

Also, the body has already been drained above...

Reply:
I could not think of a scenario where someone would response with a multi GB response with an error status code, but I see your point.